server/submission/queries.ts

```diff
@@ -1,25 +1,61 @@
-import { Submission } from 'server/models';
-import { Submission as SubmissionType, managerStatuses, submitterStatuses } from 'types';
+import { createPub } from 'server/pub/queries';
+import { Collection, Submission, SubmissionWorkflow } from 'server/models';
+import {
+	Submission as SubmissionType,
+	SequelizeModel,
+	managerStatuses,
+	submitterStatuses,
+} from 'types';
 
 const updateToStatuses = [...managerStatuses, ...submitterStatuses] as const;
 
-type UpdateToStatus = typeof updateToStatuses;
-
-type PatchType = Partial<SubmissionType> & { status: UpdateToStatus };
+type CreateOptions = {
+	userId: string;
+	submissionWorkflowId: string;
+};
 
-export const createSubmission = async ({ pubId }) =>
-	Submission.create({
-		pubId,
+export const createSubmission = async ({ userId, submissionWorkflowId }: CreateOptions) => {
+	const { collection } = await SubmissionWorkflow.findOne({
+		where: { id: submissionWorkflowId },
+		include: [{ model: Collection, as: 'collection' }],
+	});
+	const pub = await createPub(
+		{
+			communityId: collection.communityId,
+			collectionIds: [collection.id],
+		},
+		userId,
+	);
+	return Submission.create({
+		pubId: pub.id,
+		submissionWorkflowId,
 		status: 'incomplete',
 	});
+};
+
+type UpdateToStatus = typeof updateToStatuses;
+type UpdateOptions = Partial<SubmissionType> & { status: UpdateToStatus };
+
+export const updateSubmission = async (patch: UpdateOptions, sendEmail = false) => {
+	const { id, status } = patch;
+	const submission: SequelizeModel<SubmissionType> = await Submission.findOne({ where: { id } });
+	const previousStatus = submission.status;
+	submission.status = status;
+	await submission.save();
+	if (sendEmail) {
+		if (previousStatus === 'incomplete' && status === 'submitted') {
+			// Send an email from here
+		}
+	}
 
-export const updateSubmission = async (patch: PatchType) =>
 	Submission.update(
 		{ status: patch.status },
```

Comment:
why not just pass status here?

Reply:
I could see an `updateSubmissionStatus` function being an appropriate way to handle this, but we need to pass in `{id, status}` anyway so kinda it might as well be a single object. But I didn't write this function :)